### DISCUSSION TITLE
Fake Path to the file which called ray()

### DISCUSSION THREAD

**Discussion by DanielFatkic** [spatie/ray#176] [Q&A] [answered]
Is there a way to pass a specific path to the file which I want to associate with the ray call (especially to open it in the IDE) when ray() is called? 

We have some fancy system where some php code is stored in the database and which is called via eval() via our autoloader, where I'd really love to be able to pass a specific filename to ray to be able to open my specific fake file in the IDE, because right now it's only going to link to the file which contains the autoloader which handles the magic.

Something like `ray($this,'_FILE_) or ray($this)->localPath(__FILE__) .`

**Comment by DanielFatkic** [comment, marked as answer]
Issue resolved. It seems that I can use a custom Payload with an own file_name and line_number declaration for this.

**Reply by jozydapozy** [reply]
Sounds promising. Did you fork the project to get this working or is there a way to inject your custom Payload?

**Reply by slaFFik** [reply]
@jozydapozy You can fake it using macro: https://spatie.be/docs/ray/v1/advanced-usage/adding-your-own-ray-functions-in-php:
> If you want to control the little label next to the item you should use `sendCustom` in your macro.
```php
Ray::macro('myCustomFunction', function() {
    $this->sendCustom('my custom content', 'hey');
});
```
And in `hey` value you can define your custom label.

**Reply by jozydapozy** [reply]
That's really cool! Didn't spot that feature earlier..! thanks
